plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/aps/system/services/resource/ResourcesService.java

```diff
@@ -0,0 +1,336 @@
+
+            BaseResourceDataBean resourceFile = new BaseResourceDataBean();
+            resourceFile.setResourceType(resource.getType());
+            resourceFile.setResourceId(resourceId);
+            resourceFile.setMetadata(resource.getMetadata());
+
+
+            if (file != null) {
+                validateMimeType(resource.getType(), file.getContentType());
+
+                resourceFile.setInputStream(file.getInputStream());
+                resourceFile.setFileSize(file.getBytes().length / 1000);
+                resourceFile.setFileName(file.getOriginalFilename());
+                resourceFile.setMimeType(file.getContentType());
+                resourceFile.setDescr(file.getOriginalFilename());
+
+                if ((description == null || description.trim().length() == 0 ) && resource.getDescription().equals(resource.getMasterFileName())) {
+                    description = resourceFile.getFileName();
+                }
+            }
+
+            if (description != null && !description.trim().isEmpty()) {
+                resourceFile.setDescr(description.trim());
+            } else {
+                resourceFile.setDescr(resource.getDescription());
+            }
+
+            if (group != null && !group.trim().isEmpty()) {
+                resourceFile.setMainGroup(group.trim());
+            } else {
+                resourceFile.setMainGroup(resource.getMainGroup());
+            }
+
+            if (categories != null && categories.size() > 0) {
+                resourceFile.setCategories(categories.stream()
+                        .map(code -> categoryManager.getCategory(code)).collect(Collectors.toList()));
+            } else {
+                resourceFile.setCategories(resource.getCategories());
+            }
+
+            resourceManager.updateResource(resourceFile);
+            return convertResourceToDto(resource.getType(), resourceManager.loadResource(resourceId));
+        } catch (ApsSystemException e) {
+            throw new RestServerError("Error updating image resource", e);
+        } catch (IOException e) {
+            throw new RestServerError("Error reading image stream", e);
+        }
+    }
+
+    /****** Auxiliary Methods ******/
+
+    private void validateMimeType(String resourceType, String mimeType) {
+        mimeType = Optional.ofNullable(mimeType)
+                .map(t -> t.split("/")[1])
+                .orElseThrow(() -> new RestServerError("Invalid mime type", null));
+
+        List<String> allowedExtensions;
+        if ("Image".equals(resourceType)){
+            allowedExtensions = imageAllowedExtensions;
+        } else if ("Attach".equals(resourceType)){
+            allowedExtensions = fileAllowedExtensions;
+        } else throw new RestServerError(String.format("Invalid resource type %s", resourceType), null);
+
+        if (!allowedExtensions.contains(mimeType)) {
+            throw new RestServerError(String.format("Invalid mime type: %s", mimeType), null);
```

Comment:
This should get logged. At a minimum the fact that an invalid type was encountered.